dhis-2/dhis-api/src/main/java/org/hisp/dhis/scheduling/JobConfiguration.java

```diff
@@ -103,101 +109,110 @@ public class JobConfiguration extends BaseIdentifiableObject implements Secondar
   private JobParameters jobParameters;
 
   /** Indicates whether this job is currently enabled or disabled. */
-  private boolean enabled = true;
-
-  // -------------------------------------------------------------------------
-  // Internally managed properties
-  // -------------------------------------------------------------------------
+  @JsonProperty private boolean enabled = true;
 
+  @JsonProperty(access = JsonProperty.Access.READ_ONLY)
   private JobStatus jobStatus;
 
-  private transient Date nextExecutionTime;
-
+  @JsonProperty(access = JsonProperty.Access.READ_ONLY)
   private JobStatus lastExecutedStatus = JobStatus.NOT_STARTED;
 
+  /**
+   * When the job execution started last (only null when a job did never run). The name is not ideal
+   * but kept for backwards compatibility.
+   */
+  @JsonProperty(access = JsonProperty.Access.READ_ONLY)
   private Date lastExecuted;
 
-  private String lastRuntimeExecution;
+  /**
+   * When the job execution finished most recently (only null when a job never finished running
+   * yet). Can be before {@link #lastExecuted} while a new run is still in progress.
+   */
+  @JsonProperty(access = JsonProperty.Access.READ_ONLY)
+  private Date lastFinished;
```

Comment:
Do we always use the `Date` type even for new fields, to keep things consistent? Do we have any plans to use/switch to the newer `LocalDate` ?

Reply:
Good question. I used `Date` for consistency. Within persisted objects we have used `Date` for timestamps this far. I started using `Instant` in other new code or some of the `java.time` class. We should certainly move towards them. Good talking point for our agenda.